sdk/messaging/azservicebus/CHANGELOG.md

```diff
@@ -1,13 +1,15 @@
 # Release History
 
-## 1.1.5 (Unreleased)
+## 1.2.0 (Unreleased)
 
 ### Features Added
 
 ### Breaking Changes
 
 ### Bugs Fixed
 
+- Links could hang when closing, preventing recovery from completing and making a link appear stalled. (PR#19886)
```

Comment:
Do we know where/why the hang happens?  Is this a bug in go-amqp we need to fix?

Reply:
Until recently I wasn't aware that we were getting stuck here. Interestingly, this came up when I removed the Close() cancellation behavior and saw that customers were actually getting blocked there. How we get into this state is less clear.

Reply:
Do we at least have a call stack so we know where the hang happens?

Reply:
I can't seem to repro this behavior locally. We had a couple of customers that were seeing hangs when we ignored the Close() context so it was pretty clear.

Reply:
Is there enough info to know if the hang was within go-amqp?

Reply:
No, we'd need to enable go-amqp's logging for that and just the azservicebus/azeventhubs package is verbose enough. We should talk, let's see if we can repro it more easily locally.